junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/config/JupiterConfiguration.java

```diff
@@ -66,4 +67,6 @@ public interface JupiterConfiguration {
 
 	CleanupMode getDefaultTempDirCleanupMode();
 
+	TempDirFactory getDefaultTempDirFactory();
```

Comment:
Doesn't this need to be a `Supplier<TempDirFactory>` so that one instance per declaration can be created?

Reply:
Here I've taken inspiration from `getDefaultDisplayNameGenerator`. I didn't write tests yet so you're most likely right 🙂 I'll check it and confirm.

Reply:
I think it's different since `DisplayNameGenerator` does not extend `Closeable` and therefore has no defined lifecycle.

Reply:
I imagined something like the following for `DefaultJupiterConfiguration`:
```java
@Override
public Supplier<TempDirFactory> getDefaultTempDirFactorySupplier() {
	return () -> tempDirFactoryConverter.get(configurationParameters, DEFAULT_FACTORY_PROPERTY_NAME) //
			.orElse(TempDirFactory.Standard.INSTANCE);
}
```

However, this limits the usefulness of `CachingJupiterConfiguration`:

```java
@Override
public Supplier<TempDirFactory> getDefaultTempDirFactorySupplier() {
	return (Supplier<TempDirFactory>) cache.computeIfAbsent(DEFAULT_FACTORY_PROPERTY_NAME,
		key -> delegate.getDefaultTempDirFactorySupplier());
}
```

as the "heavy" calculation (checking the property + instantiation) is part of the supplier.

Probably I should introduce an alternative to `InstantiatingConfigurationParameterConverter`, i.e., a new `ParameterConverter` that returns the supplier directly.

Would that make sense or do you see a better option?

Reply:
Actually, the following in DefaultJupiterConfiguration should be sufficient:

```java
@Override
public Supplier<TempDirFactory> getDefaultTempDirFactorySupplier() {
	return tempDirFactoryConverter.get(configurationParameters, DEFAULT_FACTORY_PROPERTY_NAME) //
			.map(tempDirFactory -> (Supplier<TempDirFactory>) () -> tempDirFactory) //
			.orElse(() -> TempDirFactory.Standard.INSTANCE);
}
```

Reply:
Done in a0951ac.

Reply:
That didn't quite cut it because `tempDirFactory` was still only instantiated once but closed per declaration. I pushed 48ab4699a78a05c5d257b4cd1d90719da90bd861 to address that.

Reply:
Cool, thanks!